projects/optic/src/commands/api/write-to-file.ts

```diff
@@ -34,23 +39,69 @@ export async function writeYml(
   await fs
     .readFile(path, 'utf-8')
     .then((file) => {
-      const ops: Parameters<typeof applyOperationsToYamlString>[1] = [
-        {
-          op: 'replace',
-          value: data[OPTIC_URL_KEY],
-          path: jsonPointerHelpers.compile([OPTIC_URL_KEY]),
-        },
-      ];
-
-      if (data[OPTIC_STANDARD_KEY]) {
-        ops.push({
-          op: 'replace',
-          value: data[OPTIC_STANDARD_KEY],
-          path: jsonPointerHelpers.compile([OPTIC_STANDARD_KEY]),
-        });
-      }
-
-      return applyOperationsToYamlString(file, ops);
+      return addExtensionsToYaml(
+        file,
+        data['OPTIC_STANDARD_KEY']
+          ? {
+              [OPTIC_URL_KEY]: data[OPTIC_URL_KEY],
+              [OPTIC_STANDARD_KEY]: data[OPTIC_STANDARD_KEY]!,
+            }
+          : { [OPTIC_URL_KEY]: data[OPTIC_URL_KEY] }
+      );
     })
     .then((file) => fs.writeFile(path, file));
 }
+
+export function addExtensionsToYaml(
+  yamlContents: string,
+  extensions: { [key: `x-${string}`]: string }
+): string {
+  const parsed: FlatOpenAPIV3.Document = loadYaml(yamlContents);
+
+  try {
+    // delete anything already in spec with a different value
+    const toDelete = Object.keys(extensions).filter(
+      (extension) =>
+        parsed[extension] && parsed[extension] !== extensions[extension]
+    );
+
+    const ops: Parameters<typeof applyOperationsToYamlString>[1] = toDelete.map(
+      (extension) => ({
+        op: 'remove',
+        path: jsonPointerHelpers.compile([extension]),
+      })
+    );
+
+    const cleanedYaml = applyOperationsToYamlString(yamlContents, ops);
+    const matches = /["']{0,1}openapi["']{0,1}.*$/gm.exec(cleanedYaml);
+
+    if (matches) {
+      const indexOfOpenApiLineNewLine = cleanedYaml.indexOf(
+        '\n',
+        matches.index
+      );
+
+      const extensionLines = Object.entries(extensions).map((extension) => {
+        return `${extension[0]}: "${extension[1]}"`;
+      });
+
+      return (
+        cleanedYaml.substring(0, indexOfOpenApiLineNewLine) +
+        (extensionLines.length ? '\n' + extensionLines.join(`\n`) : '') +
+        cleanedYaml.substring(indexOfOpenApiLineNewLine)
+      );
+    } else {
+      throw new Error('openapi: 3.x.x line not found');
+    }
+  } catch (e) {
```

Comment:
wrote this with a pretty robust fallback mechanism...if we get no-ops re: `openapi` is spelled wrong or other weirdness resulting in an invalid patch, it just write it to yaml naivelly.